scvi/data/fields/_layer_field.py

```diff
@@ -84,7 +94,10 @@ def register_field(self, adata: AnnData) -> dict:
         super().register_field(adata)
         if self.correct_data_format:
             _verify_and_correct_data_format(adata, self.attr_name, self.attr_key)
-        return {self.N_CELLS_KEY: adata.n_obs, self.N_VARS_KEY: adata.n_vars}
+        return {
+            self.N_VARS_KEY: adata.n_vars,
+            self.COLUMN_NAMES_KEY: np.array(adata.var_names),
```

Comment:
np.asarray(adata.var_names) -- shouldn't allocate a new array

Reply:
also I see we are removing N_CELLS?

Reply:
I removed it since this layer field is now being used generally, not just for counts. Don't want the same named summary stat across fields and also makes less sense for protein layer field. Do you think it's necessary? Is there a reason you can't use len(adata.obs)?

Reply:
brought it back since your right about this still being breaking